nix/e2e-tests/src/k8s_tests_tester.py

```
@@ -591,10 +597,12 @@ def _test_http_endpoints(self, deployment_config: dict) -> TestResult:
 
                     nar_url = url_match.group(1).strip()
 
-                    # Fetch the NAR file
+                    # Fetch the NAR file (same transient-error tolerance).
                     if self.verbose:
                         print(f"      ✓ Fetching {nar_url}")
-                    resp = requests.get(f"{base_url}/{nar_url}", timeout=HTTP_TIMEOUT)
+                    resp = get_with_retry(
+                        lambda: requests.get(f"{base_url}/{nar_url}", timeout=HTTP_TIMEOUT)
```

Ruff (0.15.15)

[warning] Function definition does not bind loop variable `nar_url`

(B023)